File: herdr-picker-scripts/herdr-picker-focus-workspace.py
#!/usr/bin/env python3
"""Focus the workspace selected in herdr-picker."""

import json
import subprocess
import sys

context = json.load(sys.stdin)
selection = context["selections"][context["step"]]["value"]
subprocess.run(["herdr", "workspace", "focus", selection["workspace_id"]], check=True)

File: herdr-picker-workspaces/src/bin/herdr-picker-focus-workspace.rs
//! Submit command: focuses the workspace selected in herdr-picker.

use std::process::ExitCode;

use herdr_picker_sdk::{run, submit};

fn main() -> ExitCode {
    let result = submit("workspace.focus", "workspace_id", "workspace_id");
    run(env!("CARGO_BIN_NAME"), result)
}
